lib/worker/base.js

```diff
@@ -19,6 +19,35 @@ import {flags, refs, sharedWorkerTeardowns} from './state.cjs';
 import {isRunningInThread, isRunningInChildProcess} from './utils.cjs';
 
 const currentlyUnhandled = setUpCurrentlyUnhandled();
+let runner = Object.create(null);
+
+// Override process.exit with an undetectable replacement
+// to report when it is called from a test (which it should never be).
+const {apply} = Reflect;
+const realExit = process.exit;
+
+async function exit(code, forceSync = false) {
+	dependencyTracking.flush();
+	if (!forceSync) {
+		await channel.flush();
+	}
+
+	apply(realExit, process, [code]);
+}
+
+process.exit = new Proxy(realExit, {
+	apply(fn, receiver, args) {
+		channel.send({type: 'internal-error', err: serializeError('Forced exit error', false, new Error('Unexpected process.exit()'), runner.file)});
```

Comment:
I think `internal` should refer to AVA doing something wrong, but this is triggered by user code. What do you think?

Reply:
I think that's valid, although there is also an argument to be made that early termination before all tests run is an internal error. But what should this be instead?

Reply:
`process-exit` maybe? Would also allow us to print a specific message, like we do for `SIGINT`:

https://github.com/avajs/ava/blob/392ab65cd4d203643e5bb29aa7f152056fb6040b/lib/reporters/default.js#L195-L200

Reply:
Done.